homeassistant/components/whirlpool/sensor.py

```diff
@@ -0,0 +1,180 @@
+"""The Washer/Dryer Sensor for Whirlpool Appliances."""
+from __future__ import annotations
+
+from collections.abc import Callable
+from dataclasses import dataclass
+import logging
+
+from whirlpool.washerdryer import MachineState, WasherDryer
+
+from homeassistant.components.sensor import (
+    SensorDeviceClass,
+    SensorEntity,
+    SensorEntityDescription,
+    SensorStateClass,
+)
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.const import STATE_UNKNOWN
+from homeassistant.core import HomeAssistant
+from homeassistant.helpers.entity import DeviceInfo
+from homeassistant.helpers.entity_platform import AddEntitiesCallback
+from homeassistant.helpers.typing import StateType
+
+from . import WhirlpoolData
+from .const import (
+    CYCLE_FUNC,
+    DOMAIN,
+    ICON_D,
+    ICON_W,
+    MACHINE_STATE,
+    TANK_FILL,
+    WASHER_STATE,
+)
+
+_LOGGER = logging.getLogger(__name__)
+
+
+def washer_state(washer: WasherDryer) -> str | None:
+    """Determine correct states for a washer."""
+
+    machine_state = washer.get_machine_state()
+    machine_cycle = None
+    for count, func in zip(range(len(CYCLE_FUNC)), CYCLE_FUNC):
+        if func(washer):
+            machine_cycle = WASHER_STATE.get(count)
+            break
+
+    if washer.get_attribute("Cavity_OpStatusDoorOpen") == "1":
+        machine_cycle = "Door open"
+        return machine_cycle
+
+    if machine_state == MachineState.RunningMainCycle and machine_cycle:
+        return machine_cycle
+
+    return MACHINE_STATE.get(machine_state, STATE_UNKNOWN)
+
+
+@dataclass
+class WhirlpoolSensorEntityDescriptionMixin:
+    """Mixin for required keys."""
+
+    value_fn: Callable
+
+
+@dataclass
+class WhirlpoolSensorEntityDescription(
+    SensorEntityDescription, WhirlpoolSensorEntityDescriptionMixin
+):
+    """Describes Whirlpool Washer sensor entity."""
+
+
+SENSORS: tuple[WhirlpoolSensorEntityDescription, ...] = (
+    WhirlpoolSensorEntityDescription(
+        key="state",
+        name="state",
```

Comment:
Entity names should start with a capital letter:
https://developers.home-assistant.io/docs/core/entity?_highlight=name#entity-naming